### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by Alexoid1**
Bubble Sort algorithm is developed in ruby this is a algorithm that sort numbers and string from the lowest value to the heights values

### CODE REVIEW

**Review by serdg0 — CHANGES_REQUESTED** [review]
# Hello @Alexoid1 @pipe2442 !

Your code is awesome, **congratulations!** Only 2 issues to fix and you will be good to go :racing_car: :racing_car: 

After testing against the original methods I found a small difference when no block is passed in your `bubble_sort_by`. Also, I didn't found any lintern

- [ ] `bubble_sort_by`:  When **NO** block is passed, it should raise a LocalJumpError.

- [ ] Add linterns.

-[OPTIONAL] It's a good Ruby practice to name your methods/files with underscores instead of camelcase (e.g bubble_sort.rb)

[Here](https://airbrake.io/blog/ruby-exception-handling/localjumperror) is an article about this error. Hope it's helpful and if not, remember you're free to make your own research :detective: , or contact me via slack.

**Happy Coding!** :computer: :nerd_face:
**_Sergio Diaz_**

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    array2.map.with_index do |_str, i|
+      return array2 unless i != array2.length - 1
+
+      if array2[i].length > array2[i + 1].length
+        array2[i], array2[i + 1] = array2[i + 1], array2[i]
+        break
+      end
+    end
+  end
+end
```
Comment (lucasmazo32) [inline bubble_sort.rb]: - [ ] _"Now create a similar method called #bubble_sort_by which sorts an array by accepting a block. Remember to use yield inside your method definition to accomplish this."_ (Extracted from the pathwayright instructions).

For the second method, I don't see the use of the `yield`. This is a keyword that let's you extract the given block so you can use it inside your function. Using this you will be able to check if the user gave you a block, and if not you can throw the error.

Also, using the given block you will be able to sort from left to right or from right to left.

If this sounds confusing, please reach me through slack, I am more likely to respond between 9 am to 6 pm GTM -5.

**Review by lucasmazo32 — CHANGES_REQUESTED** [review]
# Changes required :wrench:

You are doing a great job so far, and the project is looking good. 

For the project to get approved, kindly implement the changes mentioned below.

### Happy coding! :computer:

## Lucas Mazo

**Review by serdg0 — APPROVED** [review]
# **Congratulations** @Alexoid1 @pipe2442  :fireworks: :100: :fireworks:
Feel free to merge.
Keep the good work in your next project!
**Happy Coding!** :computer: :nerd_face:
**_Sergio Diaz_**

**Pull request merged**